fix: BROS-592: Correctly handle `whenRole` for required controls

hlomzik opened the pull request:
Basically it's just a duplication of the similar check in `Visibility` mixin.

<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

hlomzik committed:
fix: BROS-592: Correctly handle `whenRole` for required controls

Basically it's just a duplication of the similar check in `Visibility` mixin.

hlomzik commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/19509448404)
> Successfully merged:  create mode 100644 web/libs/ui/src/lib/state-chip/state-chip.tsx

hlomzik committed:
Merge branch 'develop' into 'fb-bros-592/required-when-role'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/19509448404

nick-skriabin commented on web/libs/editor/src/mixins/Required.js:
```diff
@@ -13,6 +13,8 @@ const RequiredMixin = types
 
     return {
       validate() {
+        const whenRoles = self.whenrole ? self.whenrole.split(",") : null;
```
might worth doing `toLowerCase` here and below when you check `includes` so that user won't mess it up

hlomzik replied:
we have different visibility params like `whenChoiceValue` and `whenLabelValue` and they all are case-sensitive, so I'd leave it like this for consistency. better to keep these values strict to avoid disambiguity.

bmartel commented on web/libs/editor/src/mixins/Required.js:
```diff
@@ -32,6 +34,8 @@ const RequiredMixin = types
                 if (label && label !== self.whentagname) continue;
               }
+
+              if (whenRoles && !whenRoles.includes(reg.chatmessage?.role)) continue;
```
```
              if (whenRoles && !whenRoles.includes(reg.chatmessage?.role)) continue;
```

This feels oddly specific to include `chatmessage` in a base mixin

hlomzik committed:
Merge branch 'develop' into fb-bros-592/required-when-role

hlomzik committed:
Fix `whenRole` lookup and conditions

- use less specific `role` instead of `chatmessage.role`
- move condition out of "region-selected" one
- look for `whenrole` in parent `View` tags as well

hlomzik commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/19772352842)
> Successfully merged:  create mode 100644 web/apps/labelstudio/src/pages/Settings/StorageSettings/providers/localFiles.tsx

hlomzik committed:
Merge branch 'develop' into 'fb-bros-592/required-when-role'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/19772352842

hlomzik committed:
Fix issue

```
Failed to find the parent of AppStore@<root> at depth 2
```

hlomzik committed:
Optimize to check for roles only for relevant messages

hlomzik commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/19809168509)
> Successfully merged:  4 files changed, 35 insertions(+), 10 deletions(-)

hlomzik committed:
Merge branch 'develop' into 'fb-bros-592/required-when-role'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/19809168509

hlomzik commented:
@bmartel @nick-skriabin thanks for feedback! I changed it to less specific `role`, without `chatmessage`, I don't think we can do better given that param is called `whenRole`.
Also I improved code, fixed issues and added test for LSE.
